wire-ios-data-model/Source/ManagedObjectContext/CoreDataStack.swift

```diff
@@ -190,6 +193,42 @@ public class CoreDataStack: NSObject, ContextProvider {
         })
     }
 
+    public func setup(
+        onStartMigration: () -> Void,
+        onFailure: @escaping (Error) -> Void,
+        onCompletion: @escaping (CoreDataStack) -> Void
+    ) {
+        if needsMigration {
+            onStartMigration()
+        }
+
+        DispatchQueue.global(qos: .userInitiated).async {
+            if self.needsMessagingStoreMigration() {
+                WireLogger.localStorage.info("start migration of core data messaging store!")
+
+                do {
+                    try self.migrateMessagingStore()
+                    WireLogger.localStorage.info("finished migration of core data messaging store!")
+                } catch {
+                    WireLogger.localStorage.error("failed migration of core data messaging store!")
+                    onFailure(error)
```

Comment:
should this be called from the main thread?

Reply:
fixed